components/dashboard/ActionPlan/ActionPlanUpdateDialog.js

```diff
@@ -0,0 +1,214 @@
+import { makeStyles } from '@material-ui/styles';
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Dialog from '@material-ui/core/Dialog';
+import Divider from '@material-ui/core/Divider';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import withMobileDialog from '@material-ui/core/withMobileDialog';
+import AssignmentTurnedInIcon from '@material-ui/icons/AssignmentTurnedIn';
+import NextWeekIcon from '@material-ui/icons/NextWeek';
+import VpnKeyIcon from '@material-ui/icons/VpnKey';
+
+import { useAuth } from '../../Auth';
+import { ACTION_COLORS } from './constants';
+import { DialogTitle, DialogContent, DialogActions } from '../../DialogComponents';
+import FirebasePropTypes from '../../Firebase/PropTypes';
+
+const useStyles = makeStyles(theme => ({
+  dialog: {
+    padding: theme.spacing(4, 4, 4),
+  },
+  title: {
+    fontSize: theme.spacing(3),
+    color: theme.palette.background.dark,
+  },
+  itemLabel: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  labelContainer: {
+    marginLeft: 5,
+  },
+  input: {
+    width: theme.spacing(10),
+  },
+}));
+
+function ActionPlanUpdateDialog(props) {
+  const classes = useStyles();
+  const { userDocRef } = useAuth();
+  const { show, handleClose, actionPlan } = props;
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState();
+  const [values, setValues] = useState(actionPlan);
+  const formId = 'actionPlan';
+
+  const handleChange = event => {
+    event.persist();
+    setValues(prevValues => ({ ...prevValues, [event.target.name]: event.target.value }));
+  };
+
+  const onClose = () => {
```

Comment:
onClose you are going to need to reset the values (so that non-saved inputs don't stick around when you reopen the modal) -- `setValues({})`